### PR TITLE
Fix JSONClientError

### PR DESCRIPTION
**Pull request by derekbit**

### CODE REVIEW

File: pkg/jsonrpc/client.go [hunk 1]
```diff
@@ -63,7 +62,17 @@ func (c *Client) SendMsgWithTimeout(method string, params interface{}, timeoutIn
...
+		if resp.ErrorInfo != nil {
+			responseError.Code = RespErrorCode(resp.ErrorInfo.Code)
+			responseError.Message = RespErrorMsg(resp.ErrorInfo.Message)
+		}
+		err = JSONClientError{
+			ID:            id,
+			Method:        method,
+			Params:        params,
+			ResponseError: responseError,
+		}
```
Comment (shuo-wu) [inline pkg/jsonrpc/client.go:75]: 1. What if the error is `nil`?
2. What if the error is not a `ResponseError`? For example, it is a `json.Marshal` returned error

Comment (derekbit) [reply pkg/jsonrpc/client.go]: Updated. Thank you.

File: pkg/jsonrpc/client.go [hunk 2]
```diff
@@ -63,7 +62,18 @@ func (c *Client) SendMsgWithTimeout(method string, params interface{}, timeoutIn
...
+			errorDetail := err
+			if resp.ErrorInfo != nil {
+				errorDetail = resp.ErrorInfo
+			}
+			err = JSONClientError{
+				ID:          id,
+				Method:      method,
+				Params:      params,
+				ErrorDetail: errorDetail,
+			}
```
[the remark this answers is not in the document]
Comment (derekbit) [reply pkg/jsonrpc/client.go:75]: `ErrorDetail` can be `resp.ErrorInfo` or `err` (not `resp.ErrorInfo`).
Do we only set it to `resp.ErrorInfo`?

[the remark this answers is not in the document]
Comment (shuo-wu) [reply pkg/jsonrpc/client.go:75]: Oh sorry. This should be `ErrorDetail: err`

[the remark this answers is not in the document]
Comment (derekbit) [reply pkg/jsonrpc/client.go]: Updated. Thank you.

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**